src/ghga_datasteward_kit/s3_upload/uploader.py

```diff
@@ -59,57 +60,58 @@ async def gather(self):
 class ChunkedUploader:
     """Handler class dealing with upload functionality"""
 
-    def __init__(
+    def __init__(  # noqa: PLR0913
         self,
         *,
         input_path: Path,
         config: LegacyConfig,
         encryptor: Encryptor,
         decryptor: Decryptor,
-        upload: MultipartUpload,
+        storage: S3ObjectStorage,
+        upload_params: UploadParameters,
     ) -> None:
         self.config = config
         self.input_path = input_path
         self.encryptor = encryptor
         self.decryptor = decryptor
-        self.bucket_id = get_bucket_id(config)
+        self.storage = storage
+        self.upload_params = upload_params
         self.retry_handler = configure_retries(config)
-        self.upload = upload
         self._in_sequence_part_number = 1
         self._semaphore = asyncio.Semaphore(config.client_max_parallel_transfers)
 
-    async def encrypt_and_upload(self):
+    async def encrypt_and_upload(self) -> list[str]:
         """Delegate encryption and perform multipart upload"""
```

Comment:
```suggestion
        """Delegate encryption and perform multipart upload
        
        Returns the encrypted MD5 checksums.
        """
```
Or some similar addition to the doc string